src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,259 @@
+    public static final String COMMAND_MARK = "mark";
+    public static final String COMMAND_UNMARK = "unmark";
+    public static final String COMMAND_BYE = "bye";
+    public static final String COMMAND_EVENT = "event";
+    public static final String COMMAND_DEADLINE = "deadline";
+    public static final String COMMAND_TODO = "todo";
+    public static final File filePath = new File("src/main/java/duke/task/Hage.txt");
+
+
+    private static int itemNum;
+    private static String[] dateArray;
+    private static String[] eventArray;
+    private static final Scanner SCANNER = new Scanner(System.in);
+    private static ArrayList<Task> taskArrayList = new ArrayList<>();
+
+    public static void main(String[] args) throws DukeException, IOException {
+
+        printWelcomeMessage();
+        initTaskTable();
+
+        while (true) {
+            getNewInput();
+            saveTaskList();
+        }
+    }
+
+    private static void getNewInput() throws DukeException, IOException {
+        String inputCommand = SCANNER.nextLine();
+        executeCommand(inputCommand);
+    }
+
+    private static void executeCommand(String inputCommand) throws DukeException, IOException {
+        String[] commandArr = inputCommand.split(" ");
+        int listNum;
+        switch (commandArr[0]) {
+        case COMMAND_LIST:
+            listNum = 1;
+            if (itemNum == 0) {
+                System.out.println("List is empty");
+            }
+            else {
+                for (Task a : taskArrayList) {
+                    System.out.println(listNum++ + ". " + a);
+                }
+            }
+            break;
+        case COMMAND_MARK:
+            int markNum = Integer.parseInt(commandArr[1]) - 1;
+            try {
+                taskArrayList.get(markNum).setDone(true);
+                System.out.println("Nice! I've marked this task as done:\n" + taskArrayList.get(markNum));
+            } catch(IndexOutOfBoundsException markOutOfBounds) {
+                System.out.println("Invalid mark index!!");
+            }
+            break;
+        case COMMAND_UNMARK:
+            int unMarkNum = Integer.parseInt(commandArr[1]) - 1;
+            try {
+                taskArrayList.get(unMarkNum).setDone(false);
+                System.out.println("Nice! I've marked this task as not done yet:\n" + taskArrayList.get(unMarkNum));
+            } catch(IndexOutOfBoundsException unMarkOutOfBounds) {
+                System.out.println("Unmark index is out of bounds!!");
+            }
+            break;
+        case COMMAND_DELETE:
+            int deleteIndex = Integer.parseInt(commandArr[1]) - 1;
+            try {
+                itemNum--;
+                System.out.println("Noted. I've removed this task:\n" + taskArrayList.get(deleteIndex) +
+                        "\nNow you have " + itemNum + " tasks in the list.");
+                taskArrayList.remove(deleteIndex);
```

Comment:
`itemNum` is decremented before the item is actually deleted. What if item removal was unsuccessful? Consider [refactoring](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#structure-code-logically) these lines.